### ISSUE TITLE
Panic: the_only_name: package \"upgrade\" (isInitialPkg: true, needAnalyzeSource: true): invalid line number 526 (should be < 230)

### ISSUE THREAD

**Issue by caichao1103** [golangci/golangci-lint#5289]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [X] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [X] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [X] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### Description of the problem

Panic: the_only_name: package \"upgrade\" (isInitialPkg: true, needAnalyzeSource: true): invalid line number 526 (should be < 230)

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
v1.63.3
```

</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
level=error msg="[runner] Panic: the_only_name: package \"upgrade\" (isInitialPkg: true, needAnalyzeSource: true): invalid line number 526 (should be < 230): goroutine 118016 [running]:\nruntime/debug.Stack()\n\truntime/debug/stack.go:26 +0x5e\ngithub.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyzeSafe.func1()\n\tgithub.com/golangci/golangci-lint/pkg/goanalysis/runner_action.go:50 +0x277\npanic({0x11be420?, 0xc093345b70?})\n\truntime/panic.go:785 +0x132\ngo/token.(*File).LineStart(0xc001c6ce80?, 0x386e?)\n\tgo/token/position.go:232 +0x168\ngithub.com/golangci/golangci-lint/pkg/golinters/revive.toIssue(0xc024696d20, 0xc049eb2440)\n\tgithub.com/golangci/golangci-lint/pkg/golinters/revive/revive.go:187 +0x24c\ngithub.com/golangci/golangci-lint/pkg/golinters/revive.(*wrapper).run(0xc00190e080, 0xc0007180c0, 0xc024696d20)\n\tgithub.com/golangci/golangci-lint/pkg/golinters/revive/revive.go:156 +0x4d2\ngithub.com/golangci/golangci-lint/pkg/golinters/revive.New.func1.1(0x0?)\n\tgithub.com/golangci/golangci-lint/pkg/golinters/revive/revive.go:61 +0x37\ngithub.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyze.func3(0xc0068b9560, 0xc024696d20)\n\tgithub.com/golangci/golangci-lint/pkg/goanalysis/runner_checker.go:182 +0xeb\ngithub.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyze(0xc0068b9560)\n\tgithub.com/golangci/golangci-lint/pkg/goanalysis/runner_checker.go:208 +0xa9c\ngithub.com/golangci/golangci-lint/pkg/timeutils.(*Stopwatch).TrackStage(0xc001f3d310, {0x141c83e, 0xd}, 0xc005f47f48)\n\tgithub.com/golangci/golangci-lint/pkg/timeutils/stopwatch.go:111 +0x44\ngithub.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyzeSafe(0x0?)\n\tgithub.com/golangci/golangci-lint/pkg/goanalysis/runner_action.go:54 +0x71\ngithub.com/golangci/golangci-lint/pkg/goanalysis.(*loadingPackage).analyze.func2(0xc0068b9560)\n\tgithub.com/golangci/golangci-lint/pkg/goanalysis/runner_loadingpackage.go:83 +0xa5\ncreated by github.com/golangci/golangci-lint/pkg/goanalysis.(*loadingPackage).analyze in goroutine 3921\n\tgithub.com/golangci/golangci-lint/pkg/goanalysis/runner_loadingpackage.go:78 +0x1e9\n"
```
level=error msg="Running error: can't run linter goanalysis_metalinter\ngoanalysis_metalinter: the_only_name: package \"upgrade\" (isInitialPkg: true, needAnalyzeSource: true): invalid line number 526 (should be < 230)"

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
Hello,

To be able to process your issue as quickly as possible,
can you edit your issue to add missing elements, fix the invalid elements, and complete the sections?

| Section                                                       | Status |
|---------------------------------------------------------------|--------|
| Welcome                                                       | 🟢 |
| Description of the problem                                    | 🟠 |
| Version of golangci-lint                                      | 🔴 |
| Configuration                                                 | 🔴 |
| Go environment                                                | 🔴 |
| Verbose output of running                                     | 🟠 |
| A minimal reproducible example or link to a public repository | 🔴 |
| Validation                                                    | 🔴 |
| Supporter                                                     | 🟢 |

Legend:
- 🟢: The section is complete and valid.
- 🟠: The section is incomplete.
- 🔴: The section is missing or invalid.

**Comment by ldez** [comment]
potentially fixed by #5291 but as this issue is 90% incomplete it's not possible to check that.
